apps/exclude_ranges_totalpower.cpp

```diff
@@ -0,0 +1,204 @@
+};
+
+int read_file( const char* filename, vector<cTotalPower>& total_power_vec )
+{
+   total_power_vec.clear();
+   
+   ifstream ifile{filename};
+   if(ifile)
+      cout << "Open OK !!!";
+   else 
+      cerr << "ERROR : could not open file " << filename;
+
+   string line;      
+   while(getline(ifile,line)){ // returns false when file ends, removes \n from the text (no newline)
+//      cout << "DEBUG: " << line << endl;
+      
+      std::vector<std::string> items;
+      std::string item;
+
+      // Create an input string stream from the data string
+      std::istringstream iss(line);
+
+      // Extract items using getline with a comma delimiter
+      while (std::getline(iss, item, ' ')) {
+          items.push_back(item);
+      }       
+      if( items.size() < 5 ){      
+         cout << "WARNING : line has " << items.size() << " items -> less than 5 items -> ignored" << endl;
+         continue;
+      }
+      if( items[0] == "#" ){
+//         cout << "COMMENT LINE SKIPPED" << endl;
+         continue;
+      }
+ 
+      long int timeindex = atol(items[0].c_str());
+      double total_power = atof(items[1].c_str());
+      double median_of_medians = atof(items[2].c_str());
+      double median_of_rmsiqr  = atof(items[3].c_str());
+      double median = atof(items[4].c_str());
+      double up = median_of_medians + gThreshold*median_of_rmsiqr;
+      double down = median_of_medians - gThreshold*median_of_rmsiqr;
+      total_power_vec.push_back( cTotalPower{ timeindex, total_power, median_of_medians, median_of_rmsiqr, median, up, down } );
+
+      if( gDebug ){
+         printf("LINE : |%s|\n",line.c_str());
+         printf("     : |%ld %.4f %.4f %.4f %.4f|\n",(total_power_vec.end()-1)->timeindex,(total_power_vec.end()-1)->total_power,(total_power_vec.end()-1)->median_of_median,(total_power_vec.end()-1)->median_of_rmsiqr,(total_power_vec.end()-1)->median);
+      }
+//      printf("     : |%ld %.5f %.5f %.5f %.5f|\n",tmp.timeindex,tmp.total_power,tmp.median_of_median,tmp.median_of_rmsiqr,tmp.median);
+//      printf("     : |%s %s %s %s %s|\n",items[0].c_str(),items[1].c_str(),items[2].c_str(),items[3].c_str(),items[4].c_str());
+   }
+   
+   ifile.close();
+      
+   return total_power_vec.size();
+}
+
+void find_exclude_ranges( vector<cTotalPower>& total_power_vec, const char* outfile ) 
+{
+   int n_points = total_power_vec.size();
+   FILE* outf = fopen( outfile ,"w");
+   printf("DEBUG : n_points = %d\n",n_points);
+   printf("Exclusion ranges:\n");
+   // generate exclusion ranges :
+   int border = 10;
+   int start_range = -1;
+   int last_range_start = -1, last_range_end = -1;
+   int count_exclude_range=0;
+   for(int i=50;i<n_points;i++){  // skips first 50 (number of points in the running median) - so before running median / iqr stabilise 
+      if( total_power_vec[i].total_power > total_power_vec[i].up ){
+         if( start_range < 0 ){
+            if( last_range_end>=0 && (i-last_range_end)<10 ){
```

Comment:
Maybe the small range (10) can be parametrised? As well as the running mean length (50)